src/bootstrap/src/lib.rs

```diff
@@ -1799,10 +1799,19 @@ fn envify(s: &str) -> String {
 }
 
 pub fn find_recent_config_change_ids(current_id: usize) -> Vec<usize> {
-    let index = CONFIG_CHANGE_HISTORY
-        .iter()
-        .position(|&id| id == current_id)
-        .expect(&format!("Value `{}` was not found in `CONFIG_CHANGE_HISTORY`.", current_id));
+    if !CONFIG_CHANGE_HISTORY.contains(&current_id) {
+        // If the current change-id is greater than the most recent one,
+        // return an empty list; otherwise, return the full list.
+        if let Some(max_id) = CONFIG_CHANGE_HISTORY.iter().max() {
+            if &current_id > max_id {
+                return Vec::new();
+            }
+        }
+
+        return CONFIG_CHANGE_HISTORY.to_vec();
+    }
+
+    let index = CONFIG_CHANGE_HISTORY.iter().position(|&id| id == current_id).unwrap();
```

Comment:
Isn't this whole function just this:
```rust
CONFIG_CHANGE_HISTORY.iter().skip_while(|&id| id <= current_id).cloned().collect()
```

Reply:
No, it's not the same. `CONFIG_CHANGE_HISTORY` is basically a list of PR IDs, and there is a relatively high chance that it will not always be sequential. What we need to do is find the position of current ID and take the IDs next to it without performing comparisons. In that case, `skip_while(|&id| id <= current_id)` could return incorrect results.

Reply:
Ah! `CONFIG_CHANGE_HISTORY` needs to be sorted by merge date, not by value. I can't find that documented anywhere, can you mention that in the doc comment for the const?

Knowing that now, I don't understand why we're comparing PR numbers when the configured change-id isn't in the history. I really don't know why we'd do anything in particular. Can you explain that? Maybe as a comment?

Reply:
How https://github.com/rust-lang/rust/pull/117263/commits/684a5020cf77217b45e852d6fa80a3926dc1ebcf looks like?

Reply:
Yes, that explanation helps a lot.